src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidGetDeviceCommand.cs

```diff
@@ -46,15 +47,15 @@ protected override Task<ExitCode> InvokeInternal(ILogger logger)
             }
 
             var runner = new AdbRunner(logger);
-            string apkRequiredArchitecture;
+            IEnumerable<string> apkRequiredArchitecture;
 
-            if (!string.IsNullOrEmpty(_arguments.DeviceArchitecture))
+            if (_arguments.DeviceArchitecture.Count() != 0)
```

Comment:
`Count()` would go through the whole enumeration which is more costly than just checking if there's at least 1 element

```suggestion
            if (_arguments.DeviceArchitecture.Any())
```

Reply:
Interesting, in C++ it is const so I used to use it 😃

Reply:
So here `.Count()` is an extension method for all `Enumerable` types, no matter what they are. For many of them, it cannot be const (such as linked list) since you have to traverse the whole data structure to count the elements.

So I guess `Count()` in C++ is only defined for data structures that keep their count?

Reply:
For linked list it's const as well https://en.cppreference.com/w/cpp/container/list/size There is singly-linked list and it's an exception of that rule

Reply:
Ah, this gives me flashbacks to university times when I was looking up complexities for std 😄 

Anyway, this `Count()` from `System.Linq` is a literally doing `foreach` + `++count;` and `return count;` so beware.

Also another thing to be aware of is that enumerations in C# constructed using Linq are sort of like "queries" and every time you want their values, they evaluate again (all `Where()`, all `Select()`). Sort of like SQL views. The only way to make them evaluate once is do `.ToArray()` or `.ToList()` or similar which actually turn them into the data structure.